lib/inventory_sync/async/inventory_hosts_sync.rb

```diff
@@ -35,18 +35,14 @@ def rescue_strategy_for_self
       private
 
       def add_missing_insights_facets(uuids_hash)
-        existing_facets = InsightsFacet.where(host_id: uuids_hash.keys).pluck(:host_id, :uuid)
-        missing_facets = uuids_hash.except(*existing_facets.map(&:first)).map do |host_id, uuid|
+        all_facets = uuids_hash.map do |host_id, uuid|
           {
             host_id: host_id,
             uuid: uuid,
           }
         end
-        InsightsFacet.create(missing_facets)
 
-        existing_facets.select { |host_id, uuid| uuid.empty? }.each do |host_id, _uuid|
-          InsightsFacet.where(host_id: host_id).update_all(uuid: uuids_hash[host_id])
-        end
+        InsightsFacet.upsert_all(all_facets, unique_by: :host_id)
```

Comment:
getting error of when clicking sync in the UI: `Empty list of attributes passed`
```
11:53:43 rails.1   | 2021-11-28T11:53:43 [E|bac|8288ee33] Empty list of attributes passed (ArgumentError)
11:53:43 rails.1   |  8288ee33 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.3.7/lib/active_record/insert_all.rb:9:in `initialize'
11:53:43 rails.1   |  8288ee33 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.3.7/lib/active_record/persistence.rb:243:in `new'
11:53:43 rails.1   |  8288ee33 | /home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.3.7/lib/active_record/persistence.rb:243:in `upsert_all'
11:53:43 rails.1   |  8288ee33 | /home/vagrant/foreman_rh_cloud/lib/inventory_sync/async/inventory_hosts_sync.rb:45:in `add_missing_insights_facet
```

should we prevent if hash is empty?
```suggestion
        InsightsFacet.upsert_all(all_facets, unique_by: :host_id) unless all_facets.empty?
```

Reply:
Fixed, thanks for noticing!